resources/repository.rb

```diff
@@ -16,29 +16,51 @@
 # limitations under the License.
 #
 
-property :version,                            String, default: '9.6'
+property :version,                            [String], default: '12'
 property :enable_pgdg,                        [true, false], default: true
+property :enable_pgdg_common,                 [true, false], default: true
 property :enable_pgdg_source,                 [true, false], default: false
 property :enable_pgdg_updates_testing,        [true, false], default: false
 property :enable_pgdg_source_updates_testing, [true, false], default: false
 property :yum_gpg_key_uri, String, default: 'https://download.postgresql.org/pub/repos/yum/RPM-GPG-KEY-PGDG'
 property :apt_gpg_key_uri, String, default: 'https://download.postgresql.org/pub/repos/apt/ACCC4CF8.asc'
 
 action :add do
+  # if new_resource.version.to_f >= 10.0 then
+  #   new_resource.version = new_resource.version.to_i
+  # end
+
   case node['platform_family']
 
   when 'rhel', 'fedora', 'amazon'
+
     remote_file "/etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-#{new_resource.version}" do
       source new_resource.yum_gpg_key_uri
     end
 
+    execute 'disable-postgresql-module' do
+      command 'dnf -qy module disable postgresql'
+      action :nothing
+      only_if { node['platform_version'].to_i > 7 && platform_family?('rhel') }
+    end
```

Comment:
I think this might be a better way around this. I'm doing the same thing in the percona cookbook.
```suggestion
    execute 'dnf -y module disable postgresql' do
      only_if { node['platform_version'].to_i >= 8 }
      not_if 'dnf module list postgresql | grep -q "^postgresql.*\[x\]"'
    end
```